eZ/Bundle/EzPublishIOBundle/Tests/EventListener/StreamFileListenerTest.php

```diff
@@ -97,29 +97,39 @@ public function testRespondsToIoUri()
         );
     }
 
-    public function testRespondsToIoRequest()
+    /**
+     * @dataProvider providerForTestRespondsToIoRequest
+     */
+    public function testRespondsToIoRequest(string $uri, string $expectedFileId): void
     {
-        $uri = '/var/test/storage/images/image.png';
         $host = 'phoenix-rises.fm';
         $urlPrefix = "http://$host/var/test/storage";
         $this->configureIoUrlPrefix($urlPrefix);
         $request = $this->createRequest($uri, $host);
 
         $event = $this->createEvent($request);
 
-        $binaryFile = new BinaryFile(['mtime' => new DateTime()]);
+        $binaryFile = new BinaryFile([
+            'id' => urldecode($uri),
+            'mtime' => new DateTime(),
+        ]);
+
+        $finalUri = sprintf('http://%s%s', $host, $uri);
 
         $this->ioServiceMock
-            ->expects($this->once())
+            ->expects(self::once())
             ->method('loadBinaryFileByUri')
-            ->with(sprintf('http://%s%s', $host, $uri))
-            ->will($this->returnValue($binaryFile));
+            ->with(urldecode($finalUri))
```

Comment:
Do not use the decoding mechanism as the implementation uses. Instead, use raw value, so we can be certain about inputs and outputs, and that we do not encode / decode twice by mistake.